spackbot/handlers/pipelines.py

```diff
@@ -52,8 +50,8 @@ async def run_pipeline(event, gh):
 
     url = f"{gitlab_spack_project_url}/pipeline?ref={branch}"
     headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
-    response = requests.post(url, headers=headers)
-    result = response.json()
+
+    result = await gh.post(url, headers)
```

Comment:
I think this one should be different because it's not actually talking to GitHub.  Since it's just making a simple post request to GitLab, I think it should just do its own thing like:

```python
async with aiohttp.ClientSession() as session:
    async with session.post(url, headers=headers) as response:
        result = await response.json()
```

Otherwise I think we're sending our spackbot creds over to gitlab.spack.io when we do not need to.